crates/ruff_python_formatter/src/expression/expr_lambda.rs

```diff
@@ -1,20 +1,55 @@
+use crate::comments::dangling_comments;
 use crate::context::PyFormatContext;
 use crate::expression::parentheses::{NeedsParentheses, OptionalParentheses};
-use crate::{not_yet_implemented_custom_text, FormatNodeRule, PyFormatter};
+use crate::other::arguments::ArgumentsParentheses;
+use crate::AsFormat;
+use crate::{FormatNodeRule, PyFormatter};
+use ruff_formatter::prelude::{space, text};
 use ruff_formatter::{write, Buffer, FormatResult};
-use ruff_python_ast::node::AnyNodeRef;
+use ruff_python_ast::node::{AnyNodeRef, AstNode};
 use rustpython_parser::ast::ExprLambda;
 
 #[derive(Default)]
 pub struct FormatExprLambda;
 
 impl FormatNodeRule<ExprLambda> for FormatExprLambda {
-    fn fmt_fields(&self, _item: &ExprLambda, f: &mut PyFormatter) -> FormatResult<()> {
+    fn fmt_fields(&self, item: &ExprLambda, f: &mut PyFormatter) -> FormatResult<()> {
+        let ExprLambda {
+            range: _,
+            args,
+            body,
+        } = item;
+
+        // It's possible for some `Arguments` of `lambda`s to be assigned dangling comments.
+        //
+        // a = (
+        //     lambda  # Dangling
+        //     : 1
+        // )
+        let comments = f.context().comments().clone();
+        let dangling = comments.dangling_comments(args.as_any_node_ref());
+
+        write!(f, [text("lambda")])?;
+
+        if !args.args.is_empty() {
+            write!(
+                f,
+                [
+                    space(),
+                    args.format()
+                        .with_options(ArgumentsParentheses::SkipInsideLambda),
+                ]
+            )?;
+        }
+
         write!(
             f,
-            [not_yet_implemented_custom_text(
-                "lambda NOT_YET_IMPLEMENTED_lambda: True"
-            )]
+            [
+                text(":"),
+                space(),
+                body.format(),
+                dangling_comments(dangling)
```

Comment:
Since the dangling comments are now handled here, it is safe to override `fmt_dangling_comments` with a function that just returns `Ok(())`. 

Unrelated to this PR: What would you think (and @konstin): Should we make `fmt_dangling_comments` always return `Ok(())` and instead have the default implementation assert that are dangling comments were formatted in debug builds? The downside is that we could potentially loose dangling comments in production because of it. But it seems error-prone having to remember that it is now safe override the function.

Reply:
yes, i think we should replace the current `fmt_dangling_comments`, maybe even keep a big match which nodes are allowed to have dangling comment and which should error when they have

Reply:
:eyes:

69384521d025f595e5eb95755d8993c8f5134299